### DISCUSSION TITLE
QSL confirmation behavior on OSL image upload

### DISCUSSION THREAD

**Discussion by R1BLH** [magicbug/Cloudlog#1995] [Ideas / Feature Requests]
It would be good if OSL could be marked as received (and sent) by default, which should be configured in the user profile.
When I was working in digi modes I got about 100 electronic QSL in one week from hamlog.online. There is still no API to get QSL from there, so I download QSL manually and upload them one by one to Cloudlog. 
Each upload creates two tasks: image upload and marking OSL as received and sent (uploading QSO to hamlog.online automated by side software connected to JTDX/WSJT-X) by electronic method.

**Comment by dg9vh** [comment]
If, then should this be customizable.

**Comment by magicbug** [comment]
Not something I agree with marking by default personally even via a configuration item, of course third party systems can of course mark QSL as sent via the API.

**Reply by R1BLH** [reply]
IIt is possible that I did not express my thoughts clearly. I meant to mark the QSL received and sent by the fact of manual upload of the QSL card image (QSL management tab in QSO editing).
Now I have to manually upload the QSL image first, then open the QSO editor and manually mark the QSL as received/sent (sent, because all QSLs from this source -- hamlog.online -- are only confirmed if the sent data are the same from both QSO participants).

Perhaps a solution to this problem would be to display checkboxes in the QSL card image upload interface (mark QSL accepted/sent by default method) that will perform the specified action when the image is uploaded.
This will streamline the workflow. If the number of QSL cards is small, it is not difficult to perform two actions. But when the number is large, then optimising the process makes a difference :)

![2023-02-10_11-28-21](https://user-images.githubusercontent.com/16667173/218041830-ca646ed6-da9b-4efb-b719-6144657f0637.png)

**Reply by AndreasK79** [reply]
I find this suggestion reasonable. If a checkbox is added to the upload, then the user would click it to mark QSO as confirmed via QSL. If also left unchecked as standard, you would need to opt-in, and it only adds 1 one extra click for those that want to use it.
